src/processing/mode/android/AVD.java

```diff
@@ -48,7 +48,7 @@
   static ArrayList<String> badList;
 //  static ArrayList<String> skinList;
 
-  private String[] preferredAbi = new String[50];
+  private ArrayList<String> preferredAbi = new ArrayList<>(30);
```

Comment:
The local field type should be declared as `List<String>` but assigned with `new ArrayList` as you have it. 

But don't bother allocating a size for the `ArrayList` unless you know that the most likely case here is that there will be approx 20-30 elements.